src/lib/hooks/useFetch.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import apiClient from '@/api';
+
+const useFetch = <T>(
+  defaultValue: T,
+  url: string,
+  preProcessData: any = null,
```

Comment:
```suggestion
  preProcessData: (data:T)=>void = null,
```
이런식으로 이 인자가 함수임을 명시해주면 더 읽기 쉬울 것 같습니다!

Reply:
> 이런식으로 이 인자가 함수임을 명시해주면 더 읽기 쉬울 것 같습니다!

T가 변환후의 최종타입이니 ` null | (any) => T` 이렇게 바꾸면 되겠네요!